content/guides/autodiscovery.md

```diff
@@ -48,155 +83,253 @@ By default, the Datadog Agent includes Autodiscovery support for:
+kind: Deployment
+metadata: # Don't add templates here
+  name: apache-deployment
+spec:
+  replicas: 2
+  template:
     metadata:
-      name: apache
-      annotations:
-        service-discovery.datadoghq.com/apache.check_names: '["apache"]'
-        service-discovery.datadoghq.com/apache.init_configs: '[{}]'
-        service-discovery.datadoghq.com/apache.instances: '[{"apache_status_url": "http://%%host%%/server-status?auto"}]'
       labels:
         name: apache
+      annotations:
+        service-discovery.datadoghq.com/apache.check_names: '["apache","http_check"]'
+        service-discovery.datadoghq.com/apache.init_configs: '[{},{}]'
+        service-discovery.datadoghq.com/apache.instances: '[{"apache_status_url": "http://%%host%%/server-status?auto"},{"name": "My service", "url": "http://%%host%%", timeout: 1}]'
     spec:
       containers:
-        - name: apache
-          image: httpd
-          ports:
-            - containerPort: 80
+      - name: apache # use this as the container identifier in your annotations
+        image: httpd # NOT this
+        ports:
+        - containerPort: 80
+~~~
+
+# Reference
+
+### Template Variable Indexes
+
+For containers that have many IP addresses or expose many ports, you can tell Autodiscovery which ones to choose by appending an underscore to the template variable, followed by an integer, e.g. `%%host_0%%`, `%%port_4%%`. After inspecting the container, Autodiscovery sorts the IPs and ports **numerically and in ascending order**, so for a container that exposes ports 80, 443, and 8443, `%%port_0%%` refers to port 80. Non-indexed template variables refer to the last item in the sorted list, so in this case, `%%port%%` means port 8443.
+
+You can also add a network name suffix to the `%%host%%` variable—`%%host_bridge%%`, `%%host_swarm%%`, etc—for containers attached to multiple networks. When `%%host%%` does not have a suffix, Autodiscovery picks the container's `bridge` network IP address.
+
+### Alternate Container Identifier: Labels
+
+{: #container-labels}
+
+You can identify containers by label rather than container name or image. Just label any container `com.datadoghq.sd.check.id: <SOME_LABEL>`, and then use `<SOME_LABEL>` anywhere you'd normally use a container name or image. For example, if you label a container `com.datadoghq.sd.check.id: special-container`, Autodiscovery will apply to that container any auto-conf template that contains `special-container` in its `docker_images` list.
+
+Autodiscovery can only identify each container by label OR image/name—not both—and labels take precedence. For a container that has a `com.datadoghq.sd.check.id: special-nginx` label and runs the `nginx` image, the Agent will NOT apply templates that include only `nginx` as a container identifier.
+
+### Template Source Precedence
+
+If you provide a template for the same check type via multiple template sources, the Agent will prefer, in increasing order of preference:
+
+* Files
```

Comment:
Ok, maybe this is confusing because I put it in ascending order of precedence. I thought another engineer told me KV takes highest precedence but perhaps I misunderstood.